python-backend/app/api/mines/variances/models/variance.py

```diff
@@ -0,0 +1,66 @@
+from datetime import datetime
+
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.schema import FetchedValue
+
+from ....utils.models_mixins import AuditMixin, Base
+from app.extensions import db
+
+
+class Variance(AuditMixin, Base):
+    __tablename__ = "variance"
+    variance_id = db.Column(db.Integer, primary_key=True, server_default=FetchedValue())
+    compliance_article_id = db.Column(db.Integer,
+                                      db.ForeignKey('compliance_article.compliance_article_id'),
+                                      server_default=FetchedValue())
+    mine_guid = db.Column(UUID(as_uuid=True), db.ForeignKey('mine.mine_guid'))
+    note = db.Column(db.String(300), server_default=FetchedValue())
+    issue_date = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    received_date = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    expiry_date = db.Column(db.DateTime,
+                            nullable=False,
+                            default=datetime.strptime('9999-12-31', '%Y-%m-%d'))
+
+
+    def __repr__(self):
+        return '<Variance %r>' % self.variance_id
+
+    def json(self):
+        return {
+            'variance_id': self.variance_id,
+            'compliance_article_id': self.compliance_article_id,
+            'mine_guid': str(self.mine_guid),
+            'note': self.note,
+            'issue_date': self.issue_date.isoformat(),
+            'received_date': self.received_date.isoformat(),
+            'expiry_date': self.expiry_date.isoformat()
+        }
```

Comment:
Most likely. Maybe I'm wrong, but it's either internal API calls on the backend or API calls on the frontend because of our 'microservices,' right? So, in the essence of time (impending EOS), I went with the option I already knew I could quickly accomplish EOD yesterday.

Since this will all be pulled forward after all, I can correct course and implement whichever solution makes the most sense.

Reply:
What that is needed, yes, but variances are in the mine microservice, and you have the join table. So at worst you could return the variance with the list of document_guids.